CUJExecutor/buildProfile

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+	
+
+INPUT_FILE=${1?"ERROR: need to provide an input file and a compilation flag"}
+COMPILATION_FLAG_SEQ=${2?"ERROR: need to provide an input file and a compilation flag"}
+COMMAND_SUMMARY="/tmp/build_command_summary.txt"
+CLOUD_PROFILE="utilityProfiles/cloudProfile.prof" #location of the saved cloud profile
+EMPTY_PROFILE="utilityProfiles/empty.prof" #location of an empty profile
+STARTUP_PROFILE="utilityProfiles/startup.prof" #location of an empty profile
+CUR_PROFILE="/data/misc/profiles/cur/0/com.google.android.apps.maps/primary.prof" #location of the cloud profile on the device
+STASHING_COMPILED_ACTIONS="/tmp/manipulate_compiled_actions.txt" #Contains the actions used to compile, in case callee wants it
+BASE_PROFILE=${COMPILATION_FLAG_SEQ:0:1}
+STASHING_CREATED_PROFILE="/tmp/build_created_profile.txt"
+
+echo "" > $COMMAND_SUMMARY
+
+
+
+
+
+
+echo "Resetting the profile...."
+adb shell cmd package compile --reset com.google.android.apps.maps
+
+echo "Setting the base profile..."
+#If we're given a file name, set that to be the profile we use and exit
+if [ -f "$COMPILATION_FLAG_SEQ" ] 
+then 
+	
+	FILE_EXTENSION="${COMPILATION_FLAG_SEQ##*.}"
+	echo $FILE_EXTENSION
+	if [ "$FILE_EXTENSION" = "prof" ]
+	then
+		cp $COMPILATION_FLAG_SEQ $STASHING_CREATED_PROFILE
+		echo -n "Base: $COMPILATION_FLAG_SEQ, Actions:" > $STASHING_COMPILED_ACTIONS
+		exit 0
+	fi
+	
+fi
+
+#Otherwise, depending on the flag specified by the first character, set the base profile and build the profile based on each subsequent profile
+if [ "$BASE_PROFILE" = "e" ]
+then
+	cp $EMPTY_PROFILE $STASHING_CREATED_PROFILE
+	echo -n "Base: Empty, Actions:" > $STASHING_COMPILED_ACTIONS
+elif [ "$BASE_PROFILE" = "u" ]
+then
+	cp $CLOUD_PROFILE $STASHING_CREATED_PROFILE	
+	echo -n "Base: Cloud, Actions:" > $STASHING_COMPILED_ACTIONS
+elif [ "$BASE_PROFILE" = "s" ]
+then
+	cp $STARTUP_PROFILE $STASHING_CREATED_PROFILE	
+	echo -n "Base: Startup, Actions:" > $STASHING_COMPILED_ACTIONS
+else
+	echo "ERROR: invalid base profile"
+	exit 1
+fi
+
+
+for ((i=1; i<${#COMPILATION_FLAG_SEQ}; i++))
+do
+	COMPILATION_FLAG="${COMPILATION_FLAG_SEQ:$i:1}"
+
+	if [ "$COMPILATION_FLAG" != "c" ] && [ "$COMPILATION_FLAG" != "p" ] && [ "$COMPILATION_FLAG" != "f" ] && [ "$COMPILATION_FLAG" != "m" ] 
+	then	
+		echo "Builder actions flag: $COMPILATION_FLAG is invalid"
+		exit 1
+	fi
+
+	if [ "$COMPILATION_FLAG" = "m" ] 
+	then 
+		
+		echo "Executing preparatory actions.."
+		./runCujNTimes $INPUT_FILE p 1
+
+		echo "Waiting for prepatory actions to be added to the profile..."
+		sleep 60
+	fi
+	
+
+	echo "Pushing our working profile"
+	adb push $STASHING_CREATED_PROFILE $CUR_PROFILE
+
+	echo "Restoring profile permissions"
+	adb shell chmod a+rw $CUR_PROFILE
```

Comment:
I believe adb push messes up the file owner and permissions. May want to push to a temporary location and then do an adb shell cp to keep the existing permissions.

Reply:
messes up the permission on the device? Would chmod a+rw fix that?

Reply:
`chmod` would give access, but `adb push` changes the file owner and group to be root.  It's better to reuse the existing file permissions, which `adb shell cp` does.